src/components/DAO/DAO.tsx

```diff
@@ -1,22 +1,540 @@
+      setMembers(dataJson);
+      console.log(dataJson);
+    } catch (error) {
+      console.log("Fetching getTeamData", error);
+    }
+  };
+
+  const fetchPlatformData = async () => {
+    if ($publicClient && $balances) {
+      try {
+        const platformVersion: any = await $publicClient.readContract({
+          address: platform,
+          abi: PlatformABI,
+          functionName: "platformVersion",
+        });
+
+        const platformFees: any = await $publicClient.readContract({
+          address: platform,
+          abi: PlatformABI,
+          functionName: "getFees",
+        });
+
+        const contractData: any = await $publicClient.readContract({
+          address: platform,
+          abi: PlatformABI,
+          functionName: "getData",
+        });
+
+        const profitTotalSupply = await $publicClient.readContract({
+          address: "0x48469a0481254d5945E7E56c1Eb9861429c02f44",
+          abi: ERC20ABI,
+          functionName: "totalSupply",
+        });
+
+        const _pmTotalSupply = await $publicClient.readContract({
+          address: "0xAA3e3709C79a133e56C17a7ded87802adF23083B",
+          abi: IERC721Enumerable,
+          functionName: "totalSupply",
+        });
+
+        const _sdivTotalSupply = await $publicClient.readContract({
+          address: "0x9844a1c30462B55cd383A2C06f90BB4171f9D4bB",
+          abi: ERC20ABI,
+          functionName: "totalSupply",
+        });
+
+        const treasuryBalance = await $publicClient.readContract({
+          address: "0x48469a0481254d5945E7E56c1Eb9861429c02f44",
+          abi: ERC20ABI,
+          functionName: "balanceOf",
+          args: ["0xC82676D6025bbA6Df3585d2450EF6D0eE9b8607E"],
+        });
+
+        const multisigBalance = await $publicClient.readContract({
+          address: "0x48469a0481254d5945E7E56c1Eb9861429c02f44",
+          abi: ERC20ABI,
+          functionName: "balanceOf",
+          args: [contractData[0][6]],
+        });
+
+        //tvl
+        const totalTvl: bigint = $vaults[6].reduce(
+          (total: bigint, numero: bigint) => total + numero,
+          BigInt(0)
+        );
+        const _totalTvl = formatUnits(totalTvl, 18);
+
+        //profit Token
+        const _profitTotalSupply = formatUnits(profitTotalSupply, 18);
+        const _profitTokenPrice = Number(totalTvl) / Number(profitTotalSupply);
+        const profitMarketCap =
+          Number(_profitTokenPrice) * Number(_profitTotalSupply);
+
+        const profitToken: ProfitTokenData = {
+          price: _profitTokenPrice,
+          totalSupply: _profitTotalSupply,
+          marketCap: profitMarketCap,
+        };
+
+        setProfitTokenData(profitToken);
+
+        //sdiv token
+        setSdivTotalSupply(formatUnits(_sdivTotalSupply, 18));
+
+        //pm
+        setPmTotalSupply(formatUnits(_pmTotalSupply, 18));
+
+        //platformData
+
+        const percentageFees: string[] = platformFees.map((fee: bigint) =>
```

Comment:
dont't do unnecessary checks 
fee !== 0n --> fee

Reply:
Hi tleaf!, in this case if i dont make the check, it can return me infinite when dividing something by 0. Any other alternative?